alchemiscale_fah/compute/client.py

```diff
@@ -0,0 +1,589 @@
+        return ProjectData(**self._get(self.ws_url, f"/api/projects/{project_id}"))
+
+    def list_project_files(self, project_id) -> list[FileData]:
+        """Get a list of files associated with a project.
+
+        Parameters
+        ----------
+        project_id
+            ID of the project
+        src
+            File to download.
+        dest
+            Path to download file to.
+
+        """
+        return [
+            FileData(**i)
+            for i in self._get(
+                self.ws_url,
+                f"/api/projects/{project_id}/files",
+            )
+        ]
+
+    def create_project_file(self, project_id, src: Path, dest: Path):
+        """Upload a file to the PROJECT directory tree.
+
+        Parameters
+        ----------
+        project_id
+            ID of the project
+        src
+            File to upload.
+        dest
+            Path relative to PROJECT directory to upload to.
+
+        """
+        self._upload(
+            self.ws_url,
+            f"/api/projects/{project_id}/files/{dest}",
+            src,
+        )
+
+    def create_project_file_from_bytes(self, project_id, bytedata: bytes, dest: Path):
+        self._upload_bytes(
+            self.ws_url,
+            f"/api/projects/{project_id}/files/{dest}",
+            bytedata,
+        )
+
+    def delete_project_file(self, project_id, path):
+        """Delete a file from the PROJECT directory tree.
+
+        Parameters
+        ----------
+        project_id
+            ID of the project
+        path
+            File to delete.
+
+        """
+        self._delete(
+            self.ws_url,
+            f"/api/projects/{project_id}/files/{path}",
+        )
+
+    def get_project_file(self, project_id, src, dest):
+        """Download a file from the PROJECT directory tree.
+
+        Parameters
+        ----------
+        project_id
+            ID of the project
+        src
+            File to download.
+        dest
+            Path to download file to.
+
+        """
+        self._download(self.ws_url, f"/api/projects/{project_id}/files/{src}", dest)
+
+    def get_project_file_to_bytes(self, project_id, src):
+        return self._download_bytes(
+            self.ws_url, f"/api/projects/{project_id}/files/{src}"
+        )
+
+    def get_project_jobs(self, project_id, since: datetime = None) -> JobResults:
+        """Get all active jobs for the project.
+
+        Parameters
+        ----------
+        project_id
+            ID of the project
+        since
+            If given, only include jobs that have been updated since this time.
+
+        Returns
+        -------
+        JobResults
+
+        """
+        if since is not None:
+            since = since.isoformat()
+
+        return JobResults(
+            **self._get(self.ws_url, f"/api/projects/{project_id}/jobs", since=since)
+        )
+
+    def create_run_file(self, project_id, run_id, src: Path, dest: Path):
```

Comment:
Correct! These methods address some of the more idiosyncratic bits of the [FAH adaptive sampling API](https://docs.foldingathome.org/ws/api.html), where there are explicit endpoints for creating PROJECT files, but not RUN or CLONE files. To avoid duplicating fragile endpoint strings to make the resulting upload paths consistent on the work server, we included explicit methods for adding files associated with RUNs and CLONEs in our client here.

We currently have a few more methods in the `FahAdaptiveSamplingClient` that we make use of, but the ones we make use of we do have test coverage for. Ideally I'd strip out the methods we don't use from the client (could always pull them out of `git` history if needed later), but for new it seemed pragmatic to keep them as we build additional protocols that may need them.